packages/vite/src/index.js

```diff
@@ -0,0 +1,37 @@
+/** @returns {import('vite').Plugin} */
+export default function prefreshPlugin() {
+    return {
+        transforms: [
+            {
+                as: 'js',
+                test(path, query) {
+                    if (!/\.jsx$/.test(path)) return false;
```

Comment:
Also maybe the `x` should be optional, because lots of people put JSX in .js files

```suggestion
                    if (!/\.[tj]sx?$/.test(path)) return false;
```

Reply:
@calebeby `vite` makes a conscious effort to only support jsx inside `.jsx` or `.tsx` files. See the [JSX section in their README](https://github.com/vuejs/vite/#jsx)

Reply:
thanks guys, good catching!